cpp/include/cuspatial/experimental/geometry/box.hpp

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <cuspatial/vec_2d.hpp>
+
+namespace cuspatial {
+
+/**
+ * @addtogroup types
+ * @{
+ */
+
+/**
+ * @brief A generic axis-aligned box type.
+ *
+ * @tparam T the base type for the coordinates
+ * @tparam Vertex the vector type to use for vertices, vec_2d<T> by default
+ */
+template <typename T, typename Vertex = cuspatial::vec_2d<T>>
+class alignas(sizeof(Vertex)) box {
+ public:
+  Vertex v1;
```

Comment:
Not sure if you care, but in order for CTAD to work you either need to provide a ctor or a deduction guide. 

https://godbolt.org/z/38oYcnYjs

Reply:
Thanks. This is useful for my custom structs in the other PR.

Reply:
So is this the right deduction guide? 

```
template <typename T>
box(vec_2d<T> a, vec_2d<T> b) -> box<T, vec_2d<T>>;
```

Using the ctor approach has knock-on effects due to __host__ __device__. Results in having a host/device default ctor as well, so using the deduction guide is potentially simpler.

Reply:
Since I didn't have any CTAD compilation issues before this change, it's hard to tell if I got it right.